flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/Savepoint.java

```diff
@@ -50,16 +50,45 @@
 
     private Savepoint() {}
 
+    /**
+     * Loads an existing savepoint. Useful if you want to query, modify, or extend the state of an
+     * existing application. The savepoint will be read using the state backend defined via the
+     * clusters configuration.
+     *
+     * @param env The execution environment used to transform the savepoint.
+     * @param path The path to an existing savepoint on disk.
+     * @see #load(ExecutionEnvironment, String, StateBackend)
+     */
+    public static ExistingSavepoint load(ExecutionEnvironment env, String path) throws IOException {
```

Comment:
I think most of the code can be shared with an overloaded version of this method (by extracting a private method and calling it from both).

ditto `create()`

Reply:
This class is deprecated and will likely be dropped in 1.16, I'd rather not touch it.